dashboard/lib/services/user/password_resetter_by_username.rb

```diff
@@ -0,0 +1,34 @@
+module Services
+  module User
+    class PasswordResetterByUsername < Services::Base
+      attr_reader :username
+
+      def initialize(username: nil)
+        @username = username
+      end
+
+      def call
+        user = ::User.find_by(username: username) || ::User.new(username: username)
+        return user if user.errors.present?
```

Comment:
1. We need to validate the record first to generate the errors.
2. I don't think we need this check, as a new user record with only a username will always be invalid, and there's no need to validate a found user record if there are no changes either.
```suggestion
```

Reply:
I don't think this check is necessary, as a new user record with only a username will always be invalid, and there's no need to validate a found user record if there are no changes either.
```suggestion
        return user if user.new_record?
```

Reply:
I don't think this check is necessary, as a new user record with only a username will always be invalid, and there's no need to validate a found user record if there are no changes either.
```suggestion
```